syntax = "proto3";

service PrototypeDB2 {
  rpc Get(GetRequest) returns (GetResponse) {}
  rpc Set(SetRequest) returns (SetResponse) {}
  rpc Delete(DeleteRequest) returns (DeleteResponse) {}
  rpc GetRange(GetRangeRequest) returns (GetRangeResponse) {}
}

message GetRequest { bytes key = 1; }

message GetResponse { bytes value = 11; }

message SetRequest {
  bytes key = 1;
  bytes value = 2;
}

message SetResponse {}

message DeleteRequest { bytes key = 1; }

message DeleteResponse {}

message GetRangeRequest {
  bytes startKey = 1;
  bytes stopKey = 2;
}

message GetRangeResponse { repeated bytes values = 11; }
